release_bot/init_repo.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module initializes a repo to be used by ReleaseBot
+"""
+import yaml
+import os
+from release_bot.utils import run_command_get_output
+
+def init_repo():
+    """
+    Creates all of the required configuration script required for the ReleaseBot
+    """
+    def create_conf():
+        """
+        Create the release-conf.yaml and conf.yaml
+        """
+        conf = {}
+        conf['repository_name'] = input('Please enter the repository name:')
+        conf['repository_owner'] = input('Please enter the repository owner:')
+        print('for details on how to get github token checkout'+
+            '\n\'https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/\'')
+        conf['github_token'] = input('Please enter your valid github token:')
+        refresh_interval = input('In how many seconds would you like the'+
+            'bot to check for updates (Default 180):')
+        if refresh_interval:
+            conf['refresh_interval'] = int(refresh_interval)
+        else:
+            conf['refresh_interval'] = 180
+        is_owner_user = input('Are you the owner of the repo? (Y/n):')
+        if is_owner_user.lower()=='y' or is_owner_user=='':
+            conf['github_username'] = conf['repository_owner']
+        else:
+            conf['github_username'] = input('Please enter your github usename:')
+
+        release_conf = {}
+        cwd = os.getcwd()
+        release_conf['author_email'] = run_command_get_output(cwd, f'git config user.email')[1]
+        release_conf['author_email'] = release_conf['author_email'].replace('\'','').strip()
```

Comment:
For my credentials, I didn't have to replace anything.

Reply:
@TomasTomecek Sorry, I didn't understood this?

Reply:
Tomas probably means that he (and me neither) doesn't understand why we need the `replace('\'','')`.

Reply:
This is how the config looks like for me if I don't add the replace command 
>author_email: 'shresthagrawal.31@gmail.com
>
>  '
>author_name: 'Shresth Agrawal
>
>  '
>labels: []
>trigger_on_issue: true

Reply:
If I put this into a `release-conf.yaml` then
`yaml.safe_load(Path('release-conf.yaml').read_text())['author_email']`
results into `'shresthagrawal.31@gmail.com\n'`

* the `'` around the string just annotate that it's a string, but they are not part of the string itself, so no need for `replace('\'','')`
* yes, there is a newline, which could be removed with `strip()`, but we can't work-around all badly specified values in yaml files.

Reply:
I still didnt got it? should I use `.replace('\'','').strip()` or just `.strip()` ?
Can you please elaborate this a bit?
>we can't work-around all badly specified values in yaml files
Thanks

Reply:
I think we don't need any of those.

Reply:
But then what about the `/n` char ?

Reply:
That's what I mean with the "badly specified values".
You put the newline (`\n`) into the email/name - but the bot doesn't know whether it was intentional or not - why should it pretend to be more clever than you and strip it away?

Ideally, we would verify the config file content against some schema and let the user know that there's a bad value (because `\n` can't be part of the email), but that's for another issue/PR.

Reply:
Thanks, Now I got it!